@xen-orchestra/rest-api/src/middlewares/generic-error-handler.middleware.mts

```diff
@@ -0,0 +1,38 @@
+import { createLogger } from '@xen-orchestra/log'
+import {
+  featureUnauthorized,
+  forbiddenOperation,
+  invalidParameters,
+  noSuchObject,
+  unauthorized,
+} from 'xo-common/api-errors.js'
+import { NextFunction, Request, Response } from 'express'
+
+const log = createLogger('xo:rest-api:error-handler')
+
+// must have 4 parameters to be recognized as an error middleware by express
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export default function genericErrorHandler(error: unknown, req: Request, res: Response, _next: NextFunction) {
+  if (!(error instanceof Error)) {
+    log.error(error)
+    res.status(500).json({ error })
+    return
+  }
+
+  if (noSuchObject.is(error)) {
+    res.status(404)
+  } else if (forbiddenOperation.is(error) || featureUnauthorized.is(error)) {
```

Comment:
I think featureUnauthorized could be a 405 
https://developer.mozilla.org/fr/docs/Web/HTTP/Status/405 
https://http.cat/status/405

Reply:
From what I understand, `405` is more appropriate if you are trying to make a request with an invalid HTTP verb.
`DELETE /rest/v0/vms` (should return 405 because the request to `/rest/v0/vms` is known, but not allowed with that HTTP verb.)

> The server must generate an Allow header field in a 405 status code response.
> Allow: GET, POST, HEAD